### PR TITLE
*/*: Py3.13

### PR DESCRIPTION
**Pull request by Kangie**
A number of packages I had to change to do a world upgrade with 3.13 as python single target.

### CODE REVIEW

File: dev-python/pyqtgraph/metadata.xml [hunk 1]
```diff
@@ -10,9 +10,14 @@
...
 		Despite being written entirely in python, the library is very fast
 		due to its heavy leverage of numpy for number crunching
-		and Qt's GraphicsView framework for fast display. 
+		and Qt's GraphicsView framework for fast display.
 	</longdescription>
 	<stabilize-allarches/>
+	<use>
+		<flag name="pyside6">
+			Enable support for PySide6 bindings for Qt6.
+		</flag>
```
Comment (arthurzam) [inline dev-python/pyqtgraph/metadata.xml:19]:
```suggestion
		<flag name="pyside6">Enable support for PySide6 bindings for Qt6.</flag>
```

File: dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild [hunk 2]
```diff
@@ -23,19 +23,22 @@ S=${WORKDIR}/${PN}-${P}
 LICENSE="MIT"
 SLOT="0"
 KEYWORDS="amd64 ~arm arm64 ~riscv ~x86"
-IUSE="opengl svg"
-REQUIRED_USE="test? ( opengl svg )"
+IUSE="opengl svg pyside6 qt5 +qt6"
```
Comment (arthurzam) [inline dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild:26]: sort them please

File: dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild [hunk 3]
```diff
@@ -50,6 +53,117 @@ python_prepare_all() {
...
+	# pyqtgraph will automatically use any QT bindings it finds,
+	# patch the supported Qt versions to only include the ones we have enabled
+	local upstream_supported_qt=(
+		PySide
+		PySide2
+		PySide6
+		PyQt4
+		PyQt5
+		PyQt6
+	)
```
Comment (arthurzam) [inline dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild:66]: *Consider* to do:
```suggestion
	local upstream_supported_qt=( PySide{,2,6} PyQt{4..6} )
```

File: dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild [hunk 4]
```diff
@@ -50,6 +53,117 @@ python_prepare_all() {
...
+	awk -v qt_string="$(printf "%s\n" "${gentoo_qt[@]}")" -i inplace '
+	BEGIN {
+		i = 0
+		split(qt_string, qt_array, "\n")
+		j = 1
+	}
+	/PYSIDE = '"'"'PySide'"'"'/ {
+		i = 6 # length of upstream_supported_qt
+		for (k = 1; k <= length(qt_array); k++) {
+		print qt_array[k]
```
Comment (arthurzam) [inline dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild:97]: missing indent here inside `for`?

File: dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild [hunk 5]
```diff
@@ -50,6 +53,117 @@ python_prepare_all() {
...
+			continue
+		fi
+		if use ${use_qt,,}; then
+			gentoo_qt+=( "${qt^^} = '${qt}'" )
+		else
+			gentoo_qt+=( "${qt^^} = None" )
+		fi
+	done
+
+	awk -v qt_string="$(printf "%s\n" "${gentoo_qt[@]}")" -i inplace '
```
Comment (arthurzam) [inline dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild:88]: Uhh, all those `awk` s feel to me to be simpler to read if written in Python. On the other hand, I don't want to write them, so as long as it works...

-----

Also, does it make it easier if the `qt_string` is passed through STDIN to awk?

File: dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild (hunk 2 above)
Comment (Nowa-Ammerlaan) [inline dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild:26]: I think we should consider naming the flags `pyside6 pyqt5 +pyqt6` instead, this I think reflects better what the flag actually does and then we also have symmetry with QtPy.ebuild.

File: dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild [hunk 6]
```diff
@@ -50,6 +53,117 @@ python_prepare_all() {
...
+
+	awk -v libOrder="$(printf "%s, " "${liborder[@]}")" -i inplace '
+	BEGIN {
+		libOrder = "[" substr(libOrder, 1, length(libOrder) - 2) "]"
+	}
+	/libOrder = \[PYQT6, PYSIDE6, PYQT5, PYSIDE2\]/ {
+		sub(/\[PYQT6, PYSIDE6, PYQT5, PYSIDE2\]/, libOrder)
+	}
+	{ print }
+	' pyqtgraph/Qt/__init__.py || die "Failed to patch qt version order"
```
Comment (Nowa-Ammerlaan) [inline dev-python/pyqtgraph/pyqtgraph-0.13.7.ebuild:131]: This looks fine, but it is rather complex, can't we do the same here as in QtPy.ebuild, i.e. sed the import with `raise ImportError` if the flag is disabled?

**Pull request closed without merging**